File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic/market_data_dynamic.java
import okhttp3.*;
import com.google.gson.*;
import java.util.Map;
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;
import java.time.System;


public static Map<String, Object> fetch_all_pairs_data(pair) throws Exception {
    // GET /api/ticker
    String url = "https://exchange-api.lcx.com/api/ticker";
    OkHttpClient client = new OkHttpClient();
    Request.Builder builder = new Request.Builder().url(url);
    builder.addHeader("Content-Type", "application/json");

    Request request = builder.build();
    Response response = client.newCall(request).execute();
    String body = response.body().string();
    System.out.println(body);
    return new Gson().fromJson(body, Map.class);

}

# Workflow: FetchAllPairsData (sequence)

public static Map<String, Object> get_ethusdc_ticker(pair) throws Exception {
    // GET /api/ticker
    String url = "https://exchange-api.lcx.com/api/ticker";
    OkHttpClient client = new OkHttpClient();
    Request.Builder builder = new Request.Builder().url(url);
    builder.addHeader("Content-Type", "application/json");

    Request request = builder.build();
    Response response = client.newCall(request).execute();
    String body = response.body().string();
    System.out.println(body);
    return new Gson().fromJson(body, Map.class);

}

public static Map<String, Object> get_lcxusdc_ticker(pair) throws Exception {
    // GET /api/ticker
    String url = "https://exchange-api.lcx.com/api/ticker";
    OkHttpClient client = new OkHttpClient();
    Request.Builder builder = new Request.Builder().url(url);
    builder.addHeader("Content-Type", "application/json");

    Request request = builder.build();
    Response response = client.newCall(request).execute();
    String body = response.body().string();
    System.out.println(body);
    return new Gson().fromJson(body, Map.class);

}

File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic/market_data_dynamic.zig
const std = @import("std");


pub fn fetch_all_pairs_data(allocator: std.mem.Allocator) !void {
    // GET /api/ticker
    var client = std.http.Client{ .allocator = allocator };
    defer client.deinit();

    const url = "https://exchange-api.lcx.com/api/ticker";
    var req = try client.open(.GET, try std.Uri.parse(url), .{}, .{});
    defer req.deinit();

    try req.send();
    var body = try req.reader().readAllAlloc(allocator, 1024 * 1024);
    defer allocator.free(body);

    std.debug.print("{{}}\n", .{{body}});

}

# Workflow: FetchAllPairsData (sequence)

pub fn get_ethusdc_ticker(allocator: std.mem.Allocator) !void {
    // GET /api/ticker
    var client = std.http.Client{ .allocator = allocator };
    defer client.deinit();

    const url = "https://exchange-api.lcx.com/api/ticker";
    var req = try client.open(.GET, try std.Uri.parse(url), .{}, .{});
    defer req.deinit();

    try req.send();
    var body = try req.reader().readAllAlloc(allocator, 1024 * 1024);
    defer allocator.free(body);

    std.debug.print("{{}}\n", .{{body}});

}

pub fn get_lcxusdc_ticker(allocator: std.mem.Allocator) !void {
    // GET /api/ticker
    var client = std.http.Client{ .allocator = allocator };
    defer client.deinit();

    const url = "https://exchange-api.lcx.com/api/ticker";
    var req = try client.open(.GET, try std.Uri.parse(url), .{}, .{});
    defer req.deinit();

    try req.send();
    var body = try req.reader().readAllAlloc(allocator, 1024 * 1024);
    defer allocator.free(body);

    std.debug.print("{{}}\n", .{{body}});

}

